models/categoryDao.js

```diff
@@ -0,0 +1,25 @@
+const appDataSource = require("./dataSource");
+
+const getProductsByCategory = async (
+  categoryId,
+  orderingQuery,
+  searchQuery
+) => {
+  const result = await appDataSource.query(
+    `
+  SELECT p.id AS id, p.name AS name, p.price AS price, p.image AS image, AVG(c.score) AS score , count(c.id) AS commentCount
+  FROM products p 
+  JOIN comments c 
+  ON p.id = c.products_id 
+  WHERE p.category_id = ? ${searchQuery} 
+  GROUP BY id , image , name , price ${orderingQuery} 
+  `,
+    [categoryId]
+  );
+
+  return result;
+};
+
+module.exports = {
+  getProductsByCategory,
+};
```

Comment:
```suggestion
  const result = await appDataSource.query(`
    SELECT 
      p.id AS id, 
      p.name AS name, 
      p.price AS price, 
      p.image AS image, 
      AVG(c.score) AS score , 
      count(c.id) AS commentCount
  FROM products p 
  JOIN comments c 
  ON p.id = c.products_id 
  WHERE p.category_id = ? 
  ${searchQuery} 
  GROUP BY id , image , name , price 
  ${orderingQuery} 
  `,
    [categoryId]
  );

  return result;
};

module.exports = {
  getProductsByCategory,
};
```
필터링 잘 붙이셨네요! 고생하셨습니다.
조금 더 업그레이드 해보자면, getProductsByCateogry랑 getAllProducts랑 합쳐서 
프로덕트로 필터링 하는 부분도 ${filteringQuery}로 바꿔보실 수 있으면 가장 좋습니다!